projects/storefrontlib/cms-structure/page/page-layout/page-layout.service.ts

```diff
@@ -45,7 +46,7 @@ export class PageLayoutService {
       }),
       switchMap(({ slots, pageTemplate, breakpoint }) => {
         let result = of(slots);
-        for (const handler of getLastValueSync(this.handlers$) || []) {
+        for (const handler of this.handlers || []) {
```

Comment:
Could there be a race condition here?  (like theoretically could the value of `this.handlers` change while looping over it? ).  Could a `withLatestFrom` or another rxjs operator mitigate this?

Not convinced about this myself, asking for discussion.

Reply:
The race condition should not happen. CmsPageGuard trigger the component loading. Only when module is loaded (which provides `PAGE_LAYOUT_HANDLER`), then `PageLayoutService` can getSlot.